### PR TITLE
Improve best practices with AutoLayout

### PR DESCRIPTION
**Pull request by 3lvis**
This applies for new code. Existing code doesn't need to be rewritten for this. If AutoLayout brings more issues that what it solves we'll just change this best practice to state that.

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -202,3 +203,9 @@ Completion blocks in networking calls should be returned in the main thread.
 Completion blocks should contain the `error` instead of success/failure blocks.
 
 Use a simple NSURLSession wrapper to make things simpler, [Networking](https://github.com/3lvis/Networking) is a good candidate for this.
+
+## View layout
+
+Views should be layout using Apple's Auto Layout. No third-party frameworks are recommeded at the moment but this is open for change. Just try to use the highest abstraction that's available to you, wheter this is `UIStackView` or  `NSLayoutAnchor`.
+
+Old style layout is still an option for highly dynamic layouts such as when subclassing UICollectionViewLayout, but in general lines, first try AutoLayout.
```
Comment (elland) [inline BEST_PRACTICES.md]: For super simple layouts, like `frame = frame`, we could keep the old one in place.

File: BEST_PRACTICES.md [hunk 2]
```diff
@@ -202,3 +203,9 @@ Completion blocks in networking calls should be returned in the main thread.
 Completion blocks should contain the `error` instead of success/failure blocks.
 
 Use a simple NSURLSession wrapper to make things simpler, [Networking](https://github.com/3lvis/Networking) is a good candidate for this.
+
+## View layout
+
+Views should be layout using Apple's Auto Layout. No third-party frameworks are recommeded at the moment but this is open for change. Just try to use the highest abstraction that's available to you, wheter this is `UIStackView` or  `NSLayoutAnchor`.
```
Comment (elland) [inline BEST_PRACTICES.md:209]: I wouldn't recommend using `UIStackView` from my experience with it, maybe they're improving it for _10.0_.

Comment (3lvis) [inline BEST_PRACTICES.md:209]: Is probably as buggy as Auto Layout, so I guess that's fine.

File: BEST_PRACTICES.md (hunk 1 above)
Comment (3lvis) [inline BEST_PRACTICES.md]: If is just 4 constrains I think just using AutoLayout should be ok.

File: BEST_PRACTICES.md (hunk 2 above)
Comment (elland) [inline BEST_PRACTICES.md:209]: Is auto layout buggy? 💭

File: BEST_PRACTICES.md (hunk 1 above)
Comment (elland) [inline BEST_PRACTICES.md]: Okay!

File: BEST_PRACTICES.md (hunk 2 above)
Comment (3lvis) [inline BEST_PRACTICES.md:209]: That's what I said. 😛

**Comment by 3lvis** [comment]
I guess we just need two approvals out of 3. @marijnschilling said she'll do some reading about the topic.

I might have to put some time aside too. Hehe.

giphy

**Comment by elland** [comment]
💃

**Pull request merged**